app/src/main/java/uk/ac/ebi/atlas/search/JsonGeneSearchController.java

```diff
@@ -139,6 +142,19 @@ public Boolean isMarkerGene(@RequestParam MultiValueMap<String, String> requestP
         return markerGeneFacets != null && markerGeneFacets.size() > 0;
     }
 
+    @GetMapping(value = "/json/gene-search/organism-parts",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public Set<String> getOrganismPartBySearchTerm(@RequestParam MultiValueMap<String, String> requestParams) {
+        var geneQuery = geneIdSearchService.getGeneQueryByRequestParams(requestParams);
+        var geneIds = geneIdSearchService.search(geneQuery);
+
+        if (geneIds.isEmpty() ||  geneIds.get().isEmpty()) {
+            return ImmutableSet.of();
+        }
```

Comment:
No. we have this line in `GeneIdSearchService::searchIds`:
```java
return queryMatchesKnownIds ? Optional.of(ImmutableSet.of()) : Optional.empty();
```
and this is not handled in the service layer, so we need that check here.

Reply:
I mean that this:
```java
if (geneIds.isEmpty() ||  geneIds.get().isEmpty()) {
    return ImmutableSet.of();
}
return organismPartSearchService.search(geneIds.get());
```

Is the same as:
```java
if (geneIds.isEmpty()) {
    return ImmutableSet.of();
}
return organismPartSearchService.search(geneIds.get());
```

Because `organsimPartSearchService.search()` checks if the collection is empty (L20 of `OrganismPartSearchService.java`).

Reply:
ok, got you. Changed.